src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -82,29 +82,19 @@ public function __construct(EntityManager $em, $entityName, EventDispatcherInter
      */
     public function count()
     {
-        // TODO: remove uneccessary joins from count!
-
-        $entityId = $this->entityName . '.id';
-        $this->queryBuilder = $this->createQueryBuilder()
-            ->select('count(' . $entityId . ')');
-
-        $result = $this->queryBuilder->getQuery()->getScalarResult();
-        if (!$result) {
-            return 0;
-        }
+        $subQueryBuilder = $this->createSubQueryBuilder('COUNT(' . $this->entityName . '.id)');
 
-        // in case result has multiple results,
-        // group by separated result into multiple results,
-        // so return count of results
-        if (($temp = count($result)) > 1) {
-            $result = $temp;
-        } else {
-            // reset array indices
+        $result = $subQueryBuilder->getQuery()->getScalarResult();
+        $numResults = count($result);
+        if ($numResults > 1) {
+            return $numResults;
+        } elseif ($numResults == 1) {
             $result = array_values($result[0]);
-            $result = $result[0];
+
+            return $result[0];
```

Comment:
Is `$result` a two dimensional array? Doesn't really make sense to me like that (since you already access the first level index in the line with `array_values`).